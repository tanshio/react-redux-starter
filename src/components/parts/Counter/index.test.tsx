import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Counter } from './index'

afterEach(cleanup)

describe('Counter', () => {
  it('increment', () => {
    const { getByText, getByTestId, container, asFragment } = render(
      <Counter count={2} increment={() => {}} />
    )
    fireEvent.click(getByText(/counter/i))
    expect(getByText(/counter/i)).toHaveTextContent('3')
  })
  it('matches to the snapshot', () => {
    const { getByText, getByTestId, container, asFragment } = render(
      <Counter count={2} increment={() => {}} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
