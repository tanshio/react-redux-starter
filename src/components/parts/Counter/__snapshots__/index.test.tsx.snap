// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches to the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa goJIPt"
  >
    counter-22-2000å††
  </div>
</DocumentFragment>
`;
